src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -175,20 +183,26 @@ class SwipeableDrawer extends React.Component {
       const dx = Math.abs(currentX - this.startX);
       const dy = Math.abs(currentY - this.startY);
 
-      // If the user has moved his thumb some pixels in either direction,
-      // we can safely make an assumption about whether he was intending
-      // to swipe or scroll.
-      const isSwiping = horizontalSwipe
-        ? dx > UNCERTAINTY_THRESHOLD && dy <= UNCERTAINTY_THRESHOLD
-        : dy > UNCERTAINTY_THRESHOLD && dx <= UNCERTAINTY_THRESHOLD;
-
       // We are likely to be swiping, let's prevent the scroll event on iOS.
       if (dx > dy) {
         event.preventDefault();
       }
 
-      if (isSwiping) {
-        this.isSwiping = this.props.open ? 'closing' : 'opening';
+      const isSwiping = horizontalSwipe
+        ? dx > dy && dx > UNCERTAINTY_THRESHOLD
+        : dy > dx && dy > UNCERTAINTY_THRESHOLD;
```

Comment:
This `UNCERTAINTY_THRESHOLD` doesn't even exist on Android, I just realized that some minutes ago. You can swipe diagonally however you want, as long as the swipe _started_ near the edge (inside of the _swipe area_).

That, however, would mean that the entire "maybeSwiping vs. isSwiping" logic would need to be changed, which should be done in a new PR.

Reply:
It does when disableDiscovery is set.

Reply:
We will need it in the future to display the discoverability zone based on a timeout.

Reply:
You are right, but the threshold can be disabled if `disableDiscovery` is _not_ set, which would fix the stuck bug.